linebot/api.py

```diff
@@ -1148,6 +1149,244 @@ def get_bot_info(self, timeout=None):
 
+    def create_audience_group(self, audience_group_name, audiences=[],
+                              is_ifa=False, timeout=None):
+        """Create an audience group.
+
+        https://developers.line.biz/en/reference/messaging-api/#create-upload-audience-group
+
+        :param str audience_group_name: The audience's name
+        :param list audiences: An array of user IDs or IFAs
+        :param bool is_ifa: true | false
+        :return: audience group id
+        """
+        response = self._post(
+            '/v2/bot/audienceGroup/upload',
+            data=json.dumps({
+                "description": audience_group_name,
+                "isIfaAudience": is_ifa,
+                "audiences": [{"id": audience} for audience in audiences[:10000]],
+            }),
+            timeout=timeout
+        )
+        audience_group_id = response.json.get('audienceGroupId')
+        if 10000 < len(audiences):
+            self.add_audiences_to_audience_group(audience_group_id, audiences[10000:])
+
+        return audience_group_id
+
+    def get_audience_group(self, audience_group_id, timeout=None):
+        """Get the object of audience group.
+
+        https://developers.line.biz/en/reference/messaging-api/#get-audience-group
+
+        :param str audience_group_id: The audience ID
+        :param timeout: (optional) How long to wait for the server
+            to send data before giving up, as a float,
+            or a (connect timeout, read timeout) float tuple.
+            Default is self.http_client.timeout
+        :type timeout: float | tuple(float, float)
+        :return: AudienceGroup instance
+        """
+        response = self._get(
+            '/v2/bot/audienceGroup/{audience_group_id}'.format(
+                audience_group_id=audience_group_id),
+            timeout=timeout
+        )
+        result = dict(jobs=response.json.get('jobs', []), **response.json.get('audienceGroup', {}))
+        return AudienceGroup.new_from_json_dict(result)
+
+    def get_audience_group_list(self, page=1, description=None, status=None, size=20,
+                                include_external_public_group=None, create_route=None,
+                                timeout=None):
+        """Get data for more than one audience.
+
+        https://developers.line.biz/en/reference/messaging-api/#get-audience-groups
+
+        :param int page: The page to return when getting (paginated) results. Must be 1 or higher
+        :param str description: The name of the audience(s) to return
+        :param str status: IN_PROGRESS | READY | FAILED | EXPIRED
+        :param int size: The number of audiences per page. Default: 20, Max: 40
+        :param bool include_external_public_group: true | false
+        :param str create_route: How the audience was created.
+        :type create_route: OA_MANAGER | MESSAGING_API
+        :return: AudienceGroup instance
+        """
+        params = dict([(param, eval(param)) for param in
```

Comment:
@cryeo  -san I am not sure about the `eval()`, could you help to review this?

Is it good for the SDK?

Reply:
Sorry for the late confirmation.

I understood the intention of using `eval`, construct query parameters efficiently from method arguments, but I think that using `eval` is a bad practice as many Python communities pointed out.

Furthermore, following two parameters are not constructed correctly in current code: `include_external_public_group` and `create_route`.

Reply:
modified